hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordComparator.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.record;
+
+public interface RecordComparator {
+
+    boolean isEquals(Object value1, Object value2);
```

Comment:
minor: maybe rename to `isEqual` or `equals`? It was `isEquals` previously in `RecordFactory` but "is equals" sounds a bit weird to me.

Reply:
Hmm, I think `equals()` should be reserved to compare an instance with another instance of itself (together with `hashCode()`). But `isEqual()` sounds indeed better, I've changed it.